deposit/src/main/java/edu/unc/lib/deposit/normalize/NormalizeFileObjectsJob.java

```diff
@@ -0,0 +1,188 @@
+            }
+        }
+    }
+
+    private void wrapWithWork(Resource parent, Resource fileResc) {
+        Model model = parent.getModel();
+
+        String stagingPath = getPropertyValue(fileResc, stagingLocation);
+        String filename = stagingPath.substring(stagingPath.lastIndexOf("/") + 1);
+
+        PID filePid = PIDs.get(fileResc.getURI());
+
+        // Construct the new Work object
+        PID workPid = pidMinter.mintContentPid();
+
+        Bag workBag = model.createBag(workPid.getURI());
+        workBag.addProperty(RDF.type, Cdr.Work);
+        workBag.addProperty(CdrDeposit.label, filename);
+        workBag.addProperty(Cdr.primaryObject, fileResc);
+
+        // Add premis event for work creation
+        addPremisEvent(workPid, filePid);
+
+        // Switch parent from linking to fileObject to Work
+        Bag parentBag = model.getBag(parent);
+        unlinkChild(parentBag, fileResc);
+        parentBag.add(workBag);
+        workBag.add(fileResc);
+
+        // Transfer MODS from file to work
+        transferDescription(filePid, workPid);
+
+        // Move ACLs and other properties from file to work
+        transferProperties(fileResc, workBag.asResource());
+    }
+
+    private void unlinkChild(Bag parentBag, Resource fileResc) {
+        StmtIterator stmtIt = parentBag.getModel().listStatements(parentBag, null, fileResc);
+        Statement stmt = stmtIt.next();
+        parentBag.remove(stmt);
+        stmtIt.close();
+    }
+
+    private void transferDescription(PID filePid, PID workPid) {
+        File modsFile = new File(getDescriptionDir(), filePid.getUUID() + ".xml");
+        if (modsFile.exists()) {
+            File modsDest = new File(getDescriptionDir(), workPid.getUUID() + ".xml");
+            try {
+                Files.move(modsFile.toPath(), modsDest.toPath());
+            } catch (IOException e) {
+                failJob(e, "Failed to transfer description file {0} to work at {1}",
+                        modsFile.getAbsolutePath(), modsDest.getAbsolutePath());
+            }
+        }
+    }
+
+    private void transferProperties(Resource fileResc, Resource workResc) {
+        Model model = fileResc.getModel();
+
+        // StmtIterator stmtIt = fileResc.getModel().listStatements(parentBag, null, fileResc);
+        List<Statement> aclStmts = new ArrayList<>();
+
+        StmtIterator stmtIt = fileResc.listProperties();
+
+        while (stmtIt.hasNext()) {
+            Statement stmt = stmtIt.nextStatement();
+            Property pred = stmt.getPredicate();
+            if (!CdrAcl.NS.equals(pred.getNameSpace())) {
+                continue;
+            }
+
+            aclStmts.add(stmt);
+        }
+        stmtIt.close();
+
+        model.remove(aclStmts);
+
+        for (Statement stmt : aclStmts) {
+            model.add(workResc, stmt.getPredicate(), stmt.getObject());
+        }
+
+    }
+
+    private String getPropertyValue(Resource resc, Property property) {
```

Comment:
doesn't Jena already provide a method like this?

Reply:
it has `resource.getPropertyResource(property)` which can get you a Resource object, but getting the value of a literal object is a two step, annoyingly.